fenix/app/src/main/java/org/mozilla/fenix/components/toolbar/navbar/NavigationBarButtons.kt

```diff
@@ -0,0 +1,206 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.components.toolbar.navbar
+
+import androidx.navigation.NavController
+import mozilla.components.browser.menu.view.MenuButton
+import mozilla.components.browser.thumbnails.BrowserThumbnails
+import mozilla.components.support.base.feature.ViewBoundFeatureWrapper
+import org.mozilla.fenix.GleanMetrics.StartOnHome
+import org.mozilla.fenix.NavGraphDirections
+import org.mozilla.fenix.R
+import org.mozilla.fenix.browser.BrowserFragmentDirections
+import org.mozilla.fenix.browser.browsingmode.BrowsingMode
+import org.mozilla.fenix.browser.browsingmode.BrowsingModeManager
+import org.mozilla.fenix.components.toolbar.ToolbarMenu
+import org.mozilla.fenix.components.toolbar.interactor.BrowserToolbarInteractor
+import org.mozilla.fenix.ext.nav
+import org.mozilla.fenix.tabstray.Page
+
+/**
+ * Navigation bar button types.
+ */
+sealed class NavBarButton {
+    /**
+     * A general action button, like Home or Back buttons.
+     */
+    data class ActionButton(val iconId: Int, val descriptionResourceId: Int) : NavBarButton() {
+        var onClick: () -> Unit = {}
+        var onLongPress: () -> Unit = {}
+    }
+
+    /**
+     * A tab counter.
+     */
+    object TabCounterButton : NavBarButton() {
+        var onClick: () -> Unit = {}
+    }
+
+    /**
+     * Three dot menu button.
+     */
+    data class ThreeDotMenuButton(val menuButton: MenuButton) : NavBarButton()
+}
+
+/**
+ * Holds a collection of default buttons.
+ *
+ * Items could be copied and customized for usage across the app.
+ * Also used for building previews.
+ */
+object DefaultButtons {
```

Comment:
If we stay with the dynamic approach, could we simplify it further? Could enum classes aid us in this situation?